modular/manager/bucket_migrate_scheduler.go

```diff
@@ -384,8 +387,40 @@ func (s *BucketMigrateScheduler) pickGlobalVirtualGroupForBucketMigrate(filter *
 	return gvg, nil
 }
 
+// Calculate the staking size strategy for the target GVG
+func calculateStakingSizeStrategy(manager *ManageModular) (denom string, amount sdkmath.Int, err error) {
+	var (
+		params             *storagetypes.Params
+		stakingStorageSize uint64
+	)
+
+	if params, err = manager.baseApp.Consensus().QueryStorageParamsByTimestamp(context.Background(), time.Now().Unix()); err != nil {
+		return "", sdkmath.ZeroInt(), err
+	}
+
+	gvgMeta, err := manager.virtualGroupManager.GenerateGlobalVirtualGroupMeta(params)
+	if err != nil {
+		return "", sdkmath.ZeroInt(), err
+	}
+
+	virtualGroupParams, err := manager.baseApp.Consensus().QueryVirtualGroupParams(context.Background())
+	if err != nil {
+		return "", sdkmath.ZeroInt(), err
+	}
+	// double check
+	if gvgMeta.StakingStorageSize == 0 {
```

Comment:
If the stkaing size is insufficient, we can deposit rather than creating a new GVG. 

Please check this logic and mark a todo if you want to refine it later.